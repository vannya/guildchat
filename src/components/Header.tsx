import React from 'react';
import { RouteComponentProps, withRouter } from 'react-router';
import { MenuIcon } from '../assets/menu';
export interface IHeaderProps extends RouteComponentProps {
	isAuth: boolean;
	onLogout: () => void;
}

export const Header = withRouter(
	({ isAuth, onLogout, location }: IHeaderProps): JSX.Element => {
		return (
			<div className='header'>
				<h1>
					{location?.pathname !== '/' && (
						<button className='menu-icon'>
							<MenuIcon />
						</button>
					)}
					Guild Chat
				</h1>
				{isAuth ? (
					<button onClick={onLogout}>Log out</button>
				) : (
					<div className='loggedout' />
				)}
			</div>
		);
	}
);
