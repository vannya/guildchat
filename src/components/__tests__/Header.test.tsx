import { cleanup, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import { Header, IHeaderProps } from '../Header';

describe('Header', () => {
	const getProps = (): IHeaderProps => {
		return {
			isAuth: true,
			onLogout: jest.fn()
		};
	};

	it('should match snapshot', () => {
		const props = getProps();
		const { container } = render(<Header {...props} />);
		expect(container).toMatchSnapshot();
		cleanup();
	});

	it('should not show a Logout button if logged out', () => {
		const props = getProps();
		props.isAuth = false;
		const { container } = render(<Header {...props} />);
		const buttonPlaceholder = container.querySelectorAll('.loggedout');
		expect(buttonPlaceholder).toHaveLength(1);
		cleanup();
	});

	it('should show a Logout button if logged in', () => {
		const props = getProps();
		const { container, getAllByText } = render(<Header {...props} />);
		const button = getAllByText('Log out');
		expect(button).toHaveLength(1);
		const buttonPlaceholder = container.querySelectorAll('.loggedout');
		expect(buttonPlaceholder).toHaveLength(0);
		cleanup();
	});

	it('should log out a user', () => {
		const props = getProps();
		const { getAllByText } = render(<Header {...props} />);
		const button = getAllByText('Log out');
		expect(button).toHaveLength(1);
		userEvent.click(button[0]);
		expect(props.onLogout).toHaveBeenCalledTimes(1);
		cleanup();
	});
});
