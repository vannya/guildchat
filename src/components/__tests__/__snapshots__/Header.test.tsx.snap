// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div>
  <div
    class="header"
  >
    <h1>
      Guild Chat
    </h1>
    <button>
      Log out
    </button>
  </div>
</div>
`;
